native-authentication/src/main/java/com/okta/nativeauthentication/IdxResponseTransformer.kt

```diff
@@ -144,4 +156,24 @@ internal class RealIdxResponseTransformer : IdxResponseTransformer {
         val authenticator = firstOrNull { it.capabilities.get<Capability>() != null } ?: return null
         return authenticator.capabilities.get()
     }
+
+    private fun IdxRemediation.pollingAction(
+        resultHandler: suspend (resultProducer: suspend (InteractionCodeFlow) -> OidcClientResult<IdxResponse>) -> Unit,
+    ): (suspend () -> Unit)? {
+        val remediationCapability = capabilities.get<IdxPollRemediationCapability>()
+        val authenticatorCapability = authenticators.capability<IdxPollAuthenticatorCapability>()
+
+        // Create a poll function for the available capability.
+        val pollFunction = when {
+            remediationCapability != null -> remediationCapability::poll
+            authenticatorCapability != null -> authenticatorCapability::poll
```

Comment:
Discussed this offline, but leaving a note for easier reference. Could be worth looking into what the behavior should be if both remediationCapability and authenticatorCapability are non-null.